### PR TITLE
Signin view

### LINKED ISSUE #280: Implement the sign in publishing view layout
#### Description

Implement the initial Studio view, according to the design https://www.figma.com/file/rBjfm2tJ2Pr9M0UNTduU9v/Joystream-Atlas?node-id=3119%3A0

#### Requirements

- Matches the design and its flows (without the integration logic)
- Scales on mobile devices properly

### PR DESCRIPTION
**Pull request by drillprop**
Fix #280
Figma: https://www.figma.com/file/rBjfm2tJ2Pr9M0UNTduU9v/Joystream-Atlas?node-id=3119%3A0

### CODE REVIEW

**Comment by drillprop** [comment]
Converted to a draft for now. Found some things that I could improve

**Comment by drillprop** [comment]
To clarify - views are a little different than those in the design. 
1.  There is no create account button here because we cannot really make the user create one
2.  There is no stepper on the Sign in view  
3.  Views `Accounts with no sufficient funds` were ignored
4.  Couldn't find Terms and conditions, so I just put the lorem ipsum there
5. There is no captcha, I guess it's not needed anymore. 

If you want to test flow with a signed account you need to uncomment code inside `useEffect` in StudioView

File: src/views/studio/SignInView/SignInView.style.ts
Comment (mikkio-j) [inline src/views/studio/SignInView/SignInView.style.ts]: so the thing with the buttons is that they will have `cursor: pointer;` if there is `onClick` passed to them. So you can get rid of that and it will still have pointer cause there is `onClick` that opens modal.

File: src/views/studio/SignInView/SignInView.style.ts
Comment (mikkio-j) [inline src/views/studio/SignInView/SignInView.style.ts]: same as above. It will get pointer when we provide some action to that, or `Link` can wrap that if it's going to be a link

File: src/views/studio/SignInView/SignInView.style.ts
Comment (mikkio-j) [inline src/views/studio/SignInView/SignInView.style.ts:101]: Like that solution. It could be little bit darker so the grey text is more readable. Look:

File: src/views/studio/SignInView/SignInView.style.ts
Comment (mikkio-j) [inline src/views/studio/SignInView/SignInView.style.ts:34]: We could add some `margin-bottom` here for <360px view. Buttons will need some space on small devices

File: src/views/studio/StudioView.tsx
Comment (mikkio-j) [inline src/views/studio/StudioView.tsx]: We can get rid of that after merging #386

File: src/shared/components/Spinner/Spinner.style.ts
Comment (mikkio-j) [inline src/shared/components/Spinner/Spinner.style.ts]: I believe we don't need all these properties: `text-indent`, `position`, `transform: translateZ(0);`, `font-size`

File: src/views/studio/SelectChannelView/SelectChannelView.tsx
Comment (mikkio-j) [inline src/views/studio/SelectChannelView/SelectChannelView.tsx]: this could be a link to `channel/new`

File: src/hooks/useCheckBrowser.tsx
Comment (mikkio-j) [inline src/hooks/useCheckBrowser.tsx:3]: Just something to think about. It might be worth extending that in the future with other browsers like Brave, Vivaldi, Edge, Safari. Some of them don't have unique user-agent so it would require some more logic here

File: src/shared/components/AccountBar/AccountBar.style.ts
Comment (mikkio-j) [inline src/shared/components/AccountBar/AccountBar.style.ts]:
```suggestion
import { Button, Text, Avatar } from '@/shared/components'
```

File: src/shared/components/AccountBar/AccountBar.tsx
Comment (mikkio-j) [inline src/shared/components/AccountBar/AccountBar.tsx]:
```suggestion
import { Text, Icon } from '@/shared/components'
import {
  AccountAvatar,
  AccountInfo,
  AccountSecondary,
  AccountWrapper,
  IconWrapper,
  StyledButton,
} from './AccountBar.style'
```

File: src/shared/icons/profile.svg
Comment (mikkio-j) [inline src/shared/icons/profile.svg:1]: have the same icon in #377 named `new-channel`, we should decide on naming for that and then remove it in PR that will be merged later

File: src/views/studio/SelectMemberDialog/SelectMemberDialog.style.ts
Comment (mikkio-j) [inline src/views/studio/SelectMemberDialog/SelectMemberDialog.style.ts]: again, this can be removed after merging #386

File: src/views/studio/SignInView/SignInView.tsx
Comment (mikkio-j) [inline src/views/studio/SignInView/SignInView.tsx]: Maybe it should be a `div` with background image? Since it doesn't have semantic value - it's part of the design. Same thing with `Tile`

**Review by mikkio-j — CHANGES_REQUESTED** [review]
That's a lot of work! Great job! Found some things to improve, nothing big though 👌

File: src/views/studio/StudioView.tsx
Comment (drillprop) [reply src/views/studio/StudioView.tsx]: already fixed https://github.com/Joystream/atlas/pull/388

File: src/shared/components/Spinner/Spinner.style.ts
Comment (drillprop) [reply src/shared/components/Spinner/Spinner.style.ts]: `font-size` will be needed here. It's actually a pretty cool trick. If you want to increase the size of a spinner you just need to just adjust `font-size`. Thanks to `em` value, width, height, and the border will grow proportionally to it. 
I hope it makes sense

File: src/hooks/useCheckBrowser.tsx
Comment (drillprop) [reply src/hooks/useCheckBrowser.tsx:3]: I don't think is super necessary right now. I just leave it as it is.

**Comment by drillprop** [comment]
Converted to a draft for now. Views and general flow need to be updated. 
https://www.figma.com/file/rBjfm2tJ2Pr9M0UNTduU9v/Joystream-Atlas?node-id=6223%3A5454

File: src/shared/icons/profile.svg
Comment (drillprop) [reply src/shared/icons/profile.svg:1]: actually, it's not the same icon.  
image)
image

File: src/views/studio/SelectMembershipView/Steps/TermsStep.tsx
Comment (mikkio-j) [inline src/views/studio/SelectMembershipView/Steps/TermsStep.tsx:117]: this one doesn't have pointer cursor on hover

File: src/views/studio/SelectMembershipView/SelectMembershipView.tsx
Comment (mikkio-j) [inline src/views/studio/SelectMembershipView/SelectMembershipView.tsx]: I think we need to make it smaller on mobile devices, doesn't look good on <500px

**Review by mikkio-j — COMMENTED** [review]
Nice 👌 Two minor things I found

**Review by mikkio-j — APPROVED** [review]

**Review by kdembler — CHANGES_REQUESTED** [review]
Got my first look, sorry it was so late. I'll look at the code tomorrow, from the visuals perspective:

1. The terms and conditions text doesn't have the background from the design. The gradient looks slightly weird because of that:

(figma)
Screenshot 2021-03-29 at 19 54 59

(app)
Screenshot 2021-03-29 at 19 55 06

2. Could we make the arrow button scroll fully to the bottom? I think that'd be more useful

3. It would also be great if that scroll was smooth

File: src/components/PublishingTopbar/PublishingTopbar.style.tsx
Comment (kdembler) [inline src/components/PublishingTopbar/PublishingTopbar.style.tsx]: Rebase artifact?

File: src/config/routes.ts
Comment (kdembler) [inline src/config/routes.ts]: My mistake

```suggestion
    videos: () => 'videos',
```

File: src/config/routes.ts
Comment (kdembler) [inline src/config/routes.ts]: I don't think we need this route

File: src/shared/components/AccountBar/AccountBar.tsx
Comment (kdembler) [inline src/shared/components/AccountBar/AccountBar.tsx]: What does `blank` mean here?

File: src/shared/components/Button/Button.tsx
Comment (kdembler) [inline src/shared/components/Button/Button.tsx]: When using `target="_blank"` we need to add `rel="noopener noreferrer"`: https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever/

File: src/shared/components/Button/Button.style.ts
Comment (kdembler) [inline src/shared/components/Button/Button.style.ts]: Why do these have to be included here? They don't have anything to do with the styling

File: src/shared/components/Spinner/Spinner.tsx
Comment (kdembler) [inline src/shared/components/Spinner/Spinner.tsx]: Maybe we could have predefined sizes here and use something like `small|regular|big` for size instead?

File: src/shared/components/StudioCard/StudioCard.tsx
Comment (kdembler) [inline src/shared/components/StudioCard/StudioCard.tsx]: I think we can get rid of `channel` variant here, same for `follows`. I don't think we'll be using this for channels anytime soon

File: src/shared/components/StudioHeader/StudioHeader.stories.tsx
Comment (kdembler) [inline src/shared/components/StudioHeader/StudioHeader.stories.tsx]: I think including this in storybook may be on overkill, it seems it will be used only in sign in context

File: src/shared/icons/index.ts
Comment (kdembler) [inline src/shared/icons/index.ts]: duplicate entry

File: src/views/LayoutWithRouting.tsx
Comment (kdembler) [inline src/views/LayoutWithRouting.tsx:1]: Rebase artifact

File: src/shared/components/StudioHeader/StudioHeader.tsx
Comment (kdembler) [inline src/shared/components/StudioHeader/StudioHeader.tsx]: Likewise, I'm not sure if this needs to be in shared components, same for the card.

File: src/views/studio/SelectMembershipView/Steps/AccountStep.style.ts
Comment (kdembler) [inline src/views/studio/SelectMembershipView/Steps/AccountStep.style.ts]: Should be exported at `@/shared/components`

File: src/views/studio/SelectMembershipView/Steps/TermsStep.tsx
Comment (kdembler) [inline src/views/studio/SelectMembershipView/Steps/TermsStep.tsx]: What is `isRead` used for here? I don't see it having other purpose than `isCheckboxVisible`?

File: src/shared/components/Button/Button.tsx
Comment (kdembler) [inline src/shared/components/Button/Button.tsx]: Every button now has the following attributes in DOM:
Screenshot 2021-03-30 at 15 05 34

I think we could do a better job filtering those out

**Review by kdembler — CHANGES_REQUESTED** [review]

File: src/shared/components/AccountBar/AccountBar.tsx
Comment (drillprop) [reply src/shared/components/AccountBar/AccountBar.tsx]: Previously in the design, there was a bar with no information, no avatar, where you can click and create an account.
I'll get rid of it. It's not needed anymore.

File: src/shared/components/Button/Button.tsx
Comment (kdembler) [inline src/shared/components/Button/Button.tsx]:
```suggestion
  const as = to ? (props: LinkProps) => <Link {...props} to={to} /> : undefined
  const externalLinkProps =
```

File: src/shared/components/Button/Button.tsx
Comment (kdembler) [inline src/shared/components/Button/Button.tsx]:
```suggestion
      as={as}
```

File: src/shared/components/Button/Button.tsx
Comment (kdembler) [inline src/shared/components/Button/Button.tsx]:
```suggestion
      {...externalLinkProps}
```

**Review by kdembler — APPROVED** [review]

**Pull request merged**
